README
STM32 DQ board development, with large amount of features such as:
-	RTOS port
-	File System port
-	Serial Shell demo
-	SD/TF card driver
-	LCD driver
-	RF driver
-	Bluetooth Serial
-	Camera image sample support
-	etc.
Use Keil RealView MDK IDE

	by Librae - librae8226@gmail.com



Documentation/release_notes.txt
/*********************** (C) COPYLEFT 2010 Leafgrass *************************/

This project runs on uC/OS-II V2.52 and is based on STM32F103x(ARM Cortex-M3 Core)

Take care of the configuration of project. There're some Include Paths while linking and some Defines which cannot be modified.

Procedure of driving  a device(IMPORTANT!!!):
		1. Connect hardware correctly.
		2. Prepare DEVICE_NAME.h and DEVICE_NAME.c.
		3. Enable clock of related components.
		4. Configure its related registers.
		5. Enable related interrupt while configure NVIC.
		6. Configure related IO pins.
		7. Finish test code and begin testing.

Things to do while creationg a new task(all in task_manage.h & .c):
		1. Define copacity of task stack. Task**_STK_SIZE
		2. Define task priority. **_TASK_Prio
		3. Define task stack. extern OS_STK TASK_**_STK[**_STK_SIZE]
		4. Define exported function. void Task**(void *pdata)
			

2010-06-10( original ):
	Brief: Sucessfully port uCOS-II V2.52.

2010-06-11:
		Add code "SysTick_CounterCmd(SysTick_Counter_Enable);" in SysTick_Configuration(), solve the problem that the program can't get into
			SysTickHandler(), as below:
		/*******************************************************************************
		* Function Name  : SysTick_Config
		* Description    : Configures SysTick
		* Input          : None
		* Output         : None
		* Return         : None
		*******************************************************************************/
		void SysTick_Configuration(void)
		{
		    NVIC_SystemHandlerPriorityConfig(SystemHandler_SysTick, 0, 1);
			/* Configure HCLK clock as SysTick clock source */
			SysTick_CLKSourceConfig(SysTick_CLKSource_HCLK_Div8);
			/* SysTick interrupt each 1000 Hz with HCLK equal to 72MHz */
			SysTick_SetReload(9000);
			/* Enable the SysTick Interrupt */
			SysTick_ITConfig(ENABLE);
			SysTick_CounterCmd(SysTick_Counter_Enable); //Important!! Solve "the program can't get into SysTickHandler()!!
		}

2010-06-12( update to V1.0 ):
		1.	Add EXTI8, EXTI9 on PC8, PC9, only for EXTI test.
		2.	Modify some comments in LCD5110_lib.
		3.	(in main.c) Modifiy tasks' priority LED_TASK_Prio 1 -> 5 START_TASK_Prio 2 -> 10
		4.	(in main.c) Modify code "OSTaskDel(START_TASK_Prio);" to "OSTaskSuspend(START_TASK_Prio);"

2010-07-13
		1.Drive LTM022A69B LCD ok. Just use I/O to imitate SPI. Still cannot use internal SPI to drive it.
		2.Move LCD library of LTM022A69B into "LCD_lib.c".

2010-07-15:
		Add color bar display in LCD_test().

2010-07-16( update to V1.1 ):
		1.Solve problem about position error and color error, when display a set of 16-bit color data. Mainly resulted from CPU type,
			big endian or little endian. STM32F103RBT6 is little endian while in default set.
		2.Add Draw_img();
		3.Add colorful characters and strings, add parameters about colors in display functions.(colors are defined in LTM022A69B.h)

2010-07-17:
		1.Add comments in LCD_Init().
		2.Add parameter "color" to function Draw_dot().
		3.Add function SetCursor().
		4.Unify LCD related functions' name. Add prefix "LCD_".

2010-07-19:
		Modify data type in LCD_Lib. normal --> const. At one time, modify the pointers type to const.

2010-07-20:
		1.Correct the error that OSTimeDlyHMSM() can't delay an accurate time, by modifying SysTick_Configuration() to get an correct
		clock frequency.

2010-07-31:
		Add STM32_Init.c but is not referenced, for future use of Configuration Wizard.

2010-08-01:
		Configure SPI ok, and do some test(SPI2 tx -> SPI1 rx, display on LCD).

2010-08-03:
		1.Add SPI test code in main.c, SPI2 send data, SPI1 receive data , display info on LCD, use soft SPI.
		2.Add exported macro "USESPI" to choose if use SPI to drive LCD or not.
		3.After several days researching about hard SPI to drive LCD, failed. So switch to use a faster GPIO method, as below:
			Replace	"#define LCD_RST_H()		GPIO_SetBits(LCD_CTRL_PORT, LCD_RST)" in ST library
			with	"#define LCD_RST_H()		( LCD_CTRL_PORT->BSRR = LCD_RST )"
			for a faster LCD refresh speed.
		4.Modify name "LCD_SCI" to "LCD_SI", 'cause it means Serial Interface.
		5.Modify function name "LCD_test()"	to "LCD_Test()".

2010-08-06:
		Comment off "typedef enum {FALSE = 0, TRUE = !FALSE} bool;" in stm32f10x.type.h in order to avoid warning.

2010-08-09:
		Prepare to update to v2.0.

2010-08-10(update to v2.0):
	Brief: Use ST's FWLib3.1.2.
		1. Set project(Keil RealView MDK).
			(1) RO memory area : IROM1 -- start at 0x08000000, size 0x20000.
			(2) RW memory area : IRAM1 -- start at 0x20000000, size 0x5000.
			(3) Output setup.
			(4) Listing setup, no assembler list.
			(5) C/C++ setup.
				Preprocessor Symbols : Define -- USE_STDPERIPH_DRIVER, STM32F10X_MD.
				Include Paths : ..\;
								..\..\Libraries\CMSIS\Core\CM3;
								..\..\Libraries\STM32F10x_StdPeriph_Driver\inc;
								..\..\App;
								..\..\Driver;
								..\..\OS;
								..\..\OS\port
			(6) Linker, Use memory layout from target dialog.
			(7) Debug and Utilities, Cortex-M/R J-Link/J-Trace. Disable Trace.
		2. Modify code in "includes.h".
		3. Modify startup code, did not use the st original one. Port those used in FWLib2.0 instead.
			Of course, some code are revised for FWLib3.1.2(StdPeriph_Driver and CMSIS).
			Mainly rewrite the names of the ISR.
			The most important two are : "PendSV_Handler" and "SysTick_Handler".
									---> "OSPendSV"       and "SysTick_Handler".
									(in "os_cpu_a.asm")  (in "stm32f10x_it.c")
		4. Modify initial code for SysTick in "SysTick_Configuration()" in "CPU_Init.c" because of the changing of library.
		Note : A general problem is that program always go into "B OSStartHang" in "OSStartHighRdy" in "os_cpu_a.asm", 
				and will die in there... That's caused by program hasn't step into the correct ISR yet.

App/main.c
/************************ (C) COPYLEFT 2010 Leafgrass *************************

* File Name          : main.c
* Author             : Librae
* Version            : v0.0.2
* Last Modified Date : 11/16/2010
* Description        : main program body

* >>>>>>>>>>	Hardware	<<<<<<<<<<
*	SIGNAL:
*		PA0		---		LED2(red)
*		PA1		---		LED3(yellow)
*		PA3		---		S1
*		PA8		---		S2
*	LCD_CTRL_PORT:
*		PC8		---		LCD_RST
*		PC9		---		LCD_CS
*		PC10	---		LCD_RS
*		PC11	---		LCD_SI(soft)
*		PC12	---		LCD_SCL(soft)
*	USART:
*		PA9		---		UART1_Tx
*		PA10	---		UART1_Rx
*		PB10	---		UART3_Tx(GPS_Rx)
*		PB11	---		UART3_Rx(GPS_Tx)
*	SPI:
*		PA5		---		SPI1_SCK(SD_SCLK)
*		PA6		---		SPI1_MISO(SD_MISO)
*		PA7		---		SPI1_MOSI(SD_MOSI)
*		PB13	---		SPI2_SCK(LCD_SCL)
*		PB14	---		SPI2_MISO
*		PB15	---		SPI2_MOSI(LCD_SI)
*	PWM:
*		PB0		---		TIM3_CH3(servo)
*		PB1		---		TIM3_CH4(servo)
*	MSD:
*		PA2		---		DET
*		PA4		---		CS
*		PA5		---		SCLK
*		PA6		---		MISO
*		PA7		---		MOSI
*		
*	===============================================
*	LCD connector:
	1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16
	----------------------------------------------
	|  |  |  |  |  |  |  |  |  |  |  |  |  |  |  |
	G VCC D0 1  2  3  4  5  6  7  SI CL CS RS RST NC
*	===============================================

******************************************************************************/

/* Includes ------------------------------------------------------------------*/
#include "includes.h"

/* Private typedef -----------------------------------------------------------*/
/* Private define ------------------------------------------------------------*/
/* Private macro -------------------------------------------------------------*/
/* Private variables ---------------------------------------------------------*/
/* Private function prototypes -----------------------------------------------*/
/* Private functions ---------------------------------------------------------*/

/**
  * @brief  Main program.
  * @param  None
  * @retval None
  */
int main(void)
{
	Wait_ms(1);	//Ensure I/O is ready

	#ifdef DEBUG
		debug();	//Debug Module
	#endif

	OSInit();

	OSTaskCreate( TaskStart,	//task pointer
					(void *)0,	//parameter
					(OS_STK *)&TASK_START_STK[START_STK_SIZE-1],	//task stack top pointer
					START_TASK_Prio );	//task priority
	
	OSStart();

	return 0;
}

#ifdef  USE_FULL_ASSERT

/**
  * @brief  Reports the name of the source file and the source line number
  *   where the assert_param error has occurred.
  * @param  file: pointer to the source file name
  * @param  line: assert_param error line source number
  * @retval None
  */
void assert_failed(uint8_t* file, uint32_t line)
{ 
	/* User can add his own implementation to report the file name and line number,
	 ex: printf("Wrong parameters value: file %s on line %d\r\n", file, line) */
	
	/* Infinite loop */
	while (1)
	{
	}
}

#endif

/*********************** (C) COPYLEFT 2010 Leafgrass *************************/
